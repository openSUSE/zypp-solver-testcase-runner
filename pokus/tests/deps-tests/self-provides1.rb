#!/usr/bin/ruby


# Test addition of self-provides.


require 'zypptools/lib/package'
require 'zypptools/output/output'
include ZYppTools

require '../../src/testsuite'


packages = Array.new()

pkg = Package.new("Nuremberg")
pkg.arch = "i586"
pkg.rpmsize = 42
pkg.version = "1.2"
pkg.release = 1
pkg.summary = "A pure test package"
packages.push(pkg)

pkg = Package.new("Prague")
pkg.arch = "i586"
pkg.rpmsize = 42
pkg.version = "1.4"
pkg.release = 1
pkg.summary = "A pure test package"
pkg.deps :provides, "Prague == 1.2"
packages.push(pkg)


path = Testsuite::write_repo(:yum, packages)


Testsuite::set_arch("i586")
pool = Testsuite::read_repo("file://" + path)

Testsuite::haha2(pool).each do |res|
    puts "#{res.kind} #{res.name} #{res.edition.to_s} #{res.arch.to_s}"
    Testsuite::dump_deps(res, Dep.PROVIDES)
    puts
end

