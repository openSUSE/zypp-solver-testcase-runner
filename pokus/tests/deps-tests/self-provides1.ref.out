package Nuremberg 1.2-1 i586
  Provides: Nuremberg == 1.2-1

package Prague 1.4-1 i586
  Provides: Prague == 1.2
  Provides: Prague == 1.4-1

